WalletWasabi/BranchNBound/RecursiveCoinSelector.cs

```diff
@@ -0,0 +1,90 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BranchNBound
+{
+	public class RecursiveCoinSelector : Selector
+	{
+		private bool _optimizedCoinsFound = false;
+
+		private List<Money>? FinalCoins { get; set; }
+
+		public bool TryBranchAndBound(List<Money> coins, Money target, ulong maxTolerance, ulong toleranceIncrement, out ulong tolerance, out List<Money> finalCoins)
+		{
+			var coinsDescending = coins.OrderBy(x => x);
+			finalCoins = new List<Money>();
+			tolerance = 0;
+			var currentCoins = new List<Money>();
+
+			try
+			{
+				while (tolerance <= maxTolerance)
+				{
+					Stack<Money> pool = new Stack<Money>(coinsDescending);
+					finalCoins = SearchForCoins(currentCoins, target, tolerance, pool);
```

Comment:
```suggestion
					finalCoins = SearchForCoins(currentCoins: new List<Money>, target: target, tolerance: 0, pool: coins.OrderBy(x => x));
```
Then we don't need LOC 20-24

Reply:
For me, this is less readable, at current state it would be something like
```
var coinsDescending = coins.OrderBy(x => x);
finalCoins = new List<Money>();
Tolerance = 0UL;

try
{
   while (Tolerance <= maxTolerance)
{
      finalCoins = SearchForCoins(currentCoins: new List<Money>(), target: target, tolerance: Tolerance, pool: new Stack<Money>(coinsDescending));
```

IMO we don't earn that much